### PR TITLE
Support password authentication

### PR DESCRIPTION
**Pull request by cosmo0920**
In practice, we should support password authentication in this plugin.

### CODE REVIEW

File: test/plugin/test_out_redis.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ def test_configure
     assert_equal 'localhost', @d.instance.host
     assert_equal 6379, @d.instance.port
     assert_equal 1, @d.instance.db_number
+    assert_nil @d.instance.password
```
Comment (okkez) [inline test/plugin/test_out_redis.rb]: Add test when `password` is not `nil`

File: lib/fluent/plugin/out_redis.rb [hunk 2]
```diff
@@ -29,8 +30,13 @@ def configure(conf)
     def start
       super
 
-      @redis = Redis.new(:host => @host, :port => @port,
-                         :thread_safe => true, :db => @db_number)
+      options = {:host => @host, :port => @port,
+                 :thread_safe => true, :db => @db_number}
+      if @password
+        options.merge!(:password => @password)
+      end
```
Comment (okkez) [inline lib/fluent/plugin/out_redis.rb]: `options[:password] = @password if @password` is enough.

File: lib/fluent/plugin/out_redis.rb [hunk 3]
```diff
@@ -29,8 +30,13 @@ def configure(conf)
     def start
       super
 
-      @redis = Redis.new(:host => @host, :port => @port,
-                         :thread_safe => true, :db => @db_number)
+      options = {:host => @host, :port => @port,
+                 :thread_safe => true, :db => @db_number}
```
Comment (okkez) [inline lib/fluent/plugin/out_redis.rb]: I like following style:

```ruby
options = {
  :host => @host,
  :port => @port,
  :thread_safe => true,
  :db => @db_number
}
```

Comment (yuki24) [inline lib/fluent/plugin/out_redis.rb]: Or even better:

```ruby
options = {
  host: @host,
  port: @port,
  thread_safe: true,
  db: @db_number
}
```

Comment (cosmo0920) [inline lib/fluent/plugin/out_redis.rb]: Good point.
But `{:key => value}` Hash style is used in this plugin.
If we want to use `{key: value}`, we should use without rocket hash operator style entirely in this plugin.
So, I would like to keep it as-is for now.

File: test/plugin/test_out_redis.rb (hunk 1 above)
Comment (cosmo0920) [inline test/plugin/test_out_redis.rb]: I've added `test_configure_with_password` test case for checking `password` parameter.
How about this?

Comment (okkez) [inline test/plugin/test_out_redis.rb]: Good :+1:

File: lib/fluent/plugin/out_redis.rb (hunk 3 above)
Comment (cosmo0920) [inline lib/fluent/plugin/out_redis.rb]: And I've found that Fluentd v0.10.44 did not specify `required_ruby_version`: https://rubygems.org/gems/fluentd/versions/0.10.44
So, we should support Ruby 1.8.7 in this plugin, though.... :confounded:

**Comment by cosmo0920** [comment]
Let's merge this PR!
